client/components/ScenariosList/ScenarioCard.jsx

```diff
@@ -2,82 +2,157 @@ import React from 'react';
 import { Link } from 'react-router-dom';
 import { Button, Card } from 'semantic-ui-react';
 import PropTypes from 'prop-types';
-
+import { connect } from 'react-redux';
+import { getScenarios } from '@client/actions/scenario';
 import ConfirmAuth from '@client/components/ConfirmAuth';
 import DeletedCard from './DeletedCard';
 import './ScenariosList.css';
 
-const ScenarioCard = ({ scenario, isLoggedIn }) => {
-    const { id, title, description, deleted_at, user_is_author } = scenario;
+class ScenarioCard extends React.Component {
+    constructor(props) {
+        super(props);
+
+        const { scenario } = this.props;
+
+        this.state = {
+            scenario
+        };
+
+        this.onClick = this.onClick.bind(this);
+    }
+
+    async onClick(event, { name }) {
+        if (name === 'restore') {
+            const {
+                scenario: {
+                    id,
+                    author_id,
+                    categories,
+                    consent,
+                    description,
+                    status,
+                    title
+                }
+            } = this.state;
+            const body = JSON.stringify({
+                deleted_at: null,
+                author_id,
+                categories,
+                consent,
+                description,
+                status,
+                title
```

Comment:
The way you've set things up works well, but I'm wondering about maintainability. If we add more properties to scenarios, we'll have to remember to update this list as well. What do you think about only specifying what changes (namely `deleted_at`)? I tested the code below to show what I'm thinking:
```
           const originalScenario = this.state.scenario;
           originalScenario.deleted_at = null; 
           const {
                scenario: {
                    id
                }
            } = this.state;
            const body = JSON.stringify(originalScenario);

            const { scenario } = await (await fetch(`/api/scenarios/${id}`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body
            })).json();
```
Not required, but I'd like to know what you think about it.

Reply:
I agree with your reasoning, I think my concern was inclusion of `id` and the timestamps. If it worked for you, then I'm cool with making that change to simplify!